src/VkLayer_GF_amber_scoop/src/amber_scoop_layer.cc

```diff
@@ -0,0 +1,556 @@
+
+  // Other instance functions: (none)
+};
+
+struct DeviceData {
+  VkDevice device;
+  InstanceData* instance_data;
+
+  // Most layers must store this. Required to implement vkGetDeviceProcAddr.
+  // Should not be used otherwise; all required device function pointers
+  // should be obtained during vkCreateDevice.
+  PFN_vkGetDeviceProcAddr vkGetDeviceProcAddr;
+
+  // Other device functions:
+
+  PFN_vkQueueSubmit vkQueueSubmit;
+};
+
+using InstanceMap = gf_layers::ProtectedTinyStaleMap<void*, InstanceData>;
+using DeviceMap = gf_layers::ProtectedTinyStaleMap<void*, DeviceData>;
+
+namespace {
+
+// Read-only once initialized.
+struct AmberScoopLayerSettings {
+  bool init = false;
+  uint64_t start_draw_call = 0;
+  uint64_t draw_call_count = 0;
+  std::string amber_file_name;
+};
+
+struct GlobalData {
+  InstanceMap instance_map;
+  DeviceMap device_map;
+
+  // In vkCreateInstance, we initialize |settings| by reading environment
+  // variables while holding |settings_mutex|, after which |settings| is
+  // read-only. Thus, in instance or device functions (such as
+  // vkQueuePresentKHR) we can read |settings| without holding |settings_mutex|.
+  gf_layers::MutexType settings_mutex;
+  AmberScoopLayerSettings settings;
+};
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wexit-time-destructors"
+#pragma clang diagnostic ignored "-Wglobal-constructors"
+
+// TODO(paulthomson): The GlobalData is almost "constinit", but we would need to
+//  adjust our map data structures.
+
+// Global variables (that are not constinit) are strongly discouraged.
+// However, we should be fine as long as all global data is stored in this
+// struct.
+GlobalData global_data_;  // NOLINT(cert-err58-cpp)
+#pragma clang diagnostic pop
+
+GlobalData* get_global_data() { return &global_data_; }
+
+const std::array<VkLayerProperties, 1> kLayerProperties{{{
+    "VkLayer_GF_amber_scoop",       // layerName
+    VK_MAKE_VERSION(1U, 1U, 130U),  // specVersion NOLINT(hicpp-signed-bitwise)
+    1,                              // implementationVersion
+    "Amber scoop layer.",           // description
+}}};
+
+bool is_this_layer(const char* pLayerName) {
+  return ((pLayerName != nullptr) &&
+          strcmp(pLayerName, kLayerProperties[0].layerName) == 0);
+}
+
+void init_settings_if_needed() {
+  gf_layers::ScopedLock lock(get_global_data()->settings_mutex);
+
+  AmberScoopLayerSettings& settings = get_global_data()->settings;
+
+  if (!settings.init) {
+    get_setting_uint64("VkLayer_GF_amber_scoop_START_DRAW_CALL",
+                       "debug.gf.as.start_draw_call",
+                       &settings.start_draw_call);
+    get_setting_uint64("VkLayer_GF_amber_scoop_DRAW_CALL_COUNT",
+                       "debug.gf.as.draw_call_count",
+                       &settings.draw_call_count);
+    get_setting_string("VkLayer_GF_amber_scoop_AMBER_FILE",
+                       "debug.gf.fc.output_file", &settings.amber_file_name);
```

Comment:
fc -> as

Reply:
Oh right and also rename `output_file` to `amber_file_name`